tokenserver/assignment/sqlnode/sql.py

```diff
@@ -596,10 +608,16 @@ def unassign_node(self, service, node, timestamp=None, nodeid=None):
         )
         res.close()
 
-    def get_best_node(self, service):
+    def get_best_node(self, service, email):
         """Returns the 'least loaded' node currently available, increments the
         active count on that node, and decrements the slots currently available
         """
+        if self.lucky_user(email):
+            return (
+                self.settings.get("spanner_node_id", 800),
+                self.settings.get("spanner_entry", None)
```

Comment:
Reading both the node_id and its corresponding URL from settings seems redundant, and it might mask configuration errors (such as if we accidentally set `spanner_node_id` to a value that exists in the database as a MySQL node).

What do you think about requiring ops to add an explicit entry in the `nodes` table to represent the spanner node? (There is a `tokenserver/scripts/add_node.py` script that can be used for this purpose, for reference). This would ensure it cannot conflict with any MySQL nodes. We could then have *just* the spanner node URL in the config file, and look up the appropriate id at runtime by querying the `nodes` table.  

This would also help ensure that we respect the existing semantics of the `service` parameter, as mentioned in my comment on the tests. It's useless, but I think there's value in maintaining consistent behaviour.

Reply:
Also, defaulting to `800` here seems risky, unless something prevents creating a MySQL node in the db with id `800`.

Reply:
> We could then have just the spanner node URL in the config file, and look up the appropriate id at runtime by querying the nodes table.

Actually I'm not sure we have an appropriate index for this lookup. It may be fine to do a table-scan lookup once at load time, or maybe we could put *just* the id in the config file after inserting it into the db and finding out what it is.

Reply:
As I understand it from ops, `spanner` is already set as a node_id `800`, so we're ok with hardcoding it for now. I'll reconfirm with @Micheletto when he gets back from PTO.

I'm 100% OK with ops controlling the node table (I'm half tempted to suggest they also control the services table too based on the above comment)

Reply:
Ok, great! I reckon we should be able to put *just* the spanner node id in the config file then, and use the existing logic to reach the endpoint URL out of the database from the `nodes` table.

Reply:
Re-reading this, I see that you probably need the `spanner_entry` config option in order to support the memory backend. It still seems like a footgun to support it in the sql backend. (And TBH I don't recall why we need a separate memory backend at all versus figuring out how to use a sqlite `:memory:` database).